### PR TITLE
Replace the json module with the Python implementation from CPython

### PR DESCRIPTION
**Pull request by coolreader18**
This would give us proper streaming parsing/generation support (`json.{dump,load}`), the `JSONEncoder` and `JSONDecoder` classes, and in general better compatibility with the full API of the `json` module.

### CODE REVIEW

File: vm/src/stdlib/json.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ pub fn json_loads(obj: PyObjectRef, vm: &VirtualMachine) -> PyResult {
         let module = vm
             .get_attribute(vm.sys_module.clone(), "modules")
             .unwrap()
-            .get_item("json", vm)
+            .get_item("_serde_json", vm)
```
Comment (youknowone) [inline vm/src/stdlib/json.rs:45]: do we include this version into binary, but never using?

Comment (coolreader18) [reply vm/src/stdlib/json.rs:45]: Yeah, I suppose there really isn't a point to keeping it around; most of the functionality is in `py_serde.rs` anyway.

Comment (youknowone) [reply vm/src/stdlib/json.rs]: Not sure about the future - but If we are going to *incubate* rust native libraries, probably we can grow up the rust version of library from this one. Is this not related to `_json` in CPython?

Comment (coolreader18) [reply vm/src/stdlib/json.rs]: No, `_json` has low-level json operations, e.g. "find where this string ends", "convert this string to JSON", or streaming encoding/lexing classes.

Comment (coolreader18) [reply vm/src/stdlib/json.rs]: We could find another json library, but serde_json doesn't expose any lexing functionality.

**Pull request merged**